internals/overlord/overlord.go

```diff
@@ -49,6 +49,25 @@ var (
 	defaultCachedDownloads = 5
 )
 
+// Extension represents an extension of the Overlord.
+type Extension interface {
+	// ExtraManagers is called when building an Overlord.
```

Comment:
I don't think this is quite enough of a comment. Perhaps:

```suggestion
	// ExtraManagers is called when building an Overlord, allowing additional
	// managers to be created and added to the overlord.
```